avahi-core/server.c

```diff
@@ -1606,18 +1608,28 @@ AvahiServerConfig* avahi_server_config_init(AvahiServerConfig *c) {
 }
 
 void avahi_server_config_free(AvahiServerConfig *c) {
+    AvahiAddressList *l, *next;
     assert(c);
 
     avahi_free(c->host_name);
     avahi_free(c->domain_name);
     avahi_string_list_free(c->browse_domains);
     avahi_string_list_free(c->allow_interfaces);
     avahi_string_list_free(c->deny_interfaces);
+    for (l = c->allow_addresses; l; l = next) {
+        next = l->address_next;
+        avahi_free(l);
+    }
+    for (l = c->deny_addresses; l; l = next) {
+        next = l->address_next;
+        avahi_free(l);
+    }
 }
```

Comment:
I would prefer to have static function ``avahi_address_list_free()`` defined above this function. Called twice just like avahi_string_list_free functions above.